README
DateTime/Event/Sunrise version 0.0507
=====================================

This module will return a DateTime Object for sunrise and sunset for a
given day.

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install

or use your favorite utility: CPAN shell, cpanplus, cpanminus, dist-zilla, etc.

Note: do not be disturbed by warnings "sun never rises" or "sun never sets" during
"make test". They are nothing more than warnings.

DEPENDENCIES

This module requires these other modules and libraries:

  DateTime.pm
  DateTime::Set
  DateTime::Duration
  DateTime::Span
  DateTime::SpanSet
  POSIX
  Math::Trig
  Carp
  Params::Validate
  Set::Infinite
  Test::More
  Test::Exception

COPYRIGHT AND LICENCE

Copyright © 2003, 2013, 2015, 2020 Ron Hill and Jean Forget

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl 5.16.3. For more details, see the full
text of the licenses in the LICENSE file.

This program is distributed in the hope that it will be useful, but it
is provided “as is” and without any express or implied warranties. For
details, see the full text of the licenses in the file LICENSE.

Based on a public domain C program sunriset.c (c) Paul Schlyter, 1989, 1992


Changes
Revision history for Perl extension DateTime::Event::Sunrise.

0.0507  ...
        - xxx

0.0506  Thu 2020-07-09
        - add trace statements to better understand how the computation works.
        - overhaul the precise algorithm.
        - check the precise algorithm against data from the NOAA solar calculator and from Stellarium (ticket 132336)
        - a better fix for ticket 110016, to make the algorithm really converge toward the proper value.

0.0505  Mon 2015-12-07
        - Fix ticket 110016, to prevent endless loops
          (thank you https://geocoder.opencagedata.com/)
          Actually a stop-gap measure, until the overhaul of the precise algorithm.

0.0504  Thu 2014-07-24
        - add the "is_polar_night", "is_polar_day" and "is_day_and_night" boolean methods,
          one half of fixing ticket RT91783
        - add a "silent" build parameter, which removes the warnings about the sun never rising / never setting
          the other half of fixing ticket RT91783
        - add a test file for the parameter checking

0.0503  Thu 2013-10-17
        - stylistic and kwalitee issues: license, dependencies, use warnings
        - fix typos in the comments and POD and use UTF-8
        - fix _following_sunrise, _previous_sunrise, _following_sunset, _previous_sunset (ticket 36352)
        - deprecate "iteration" keyword, replaced by "precise" (ticket 77710)
        - allow any height, not only the standard ones (ticket 8065)
        - add an "upper_limb" parameter
        - some test data are computed by a C program based on Paul Schlyter's code

0.0502  Mon 2013-07-08
        - fixed computation error visible on 19, 20, 21 and 22 March (tickets 34770, 55762 and 75927)
        - improve object class check (ticket 7605)
        - a few words in POD about polar night and midnight sun

0.0501  Wed Mar 31 2004
        - added fix for DateTime-Set after 0.1402
        - I had to remove one test (the span set test) I need to check
          to see what is going on I think I need to rewrite this test

0.05    Fri Jan 9 2004
       - Added sunrise_sunset_span method. This will return rise/set time
         for one day (as a span object). Request # 4785 on rt.cpan.org
         Added sunrise_datetime, sunset_datetime method.

0.0402  Tues, Nov 25 2003
       - Changed the convert_hour to convert to seconds (as a duration)
         Then add the duration to the day. This will provide seconds for
         the DateTime Object.
       - Added many tests (Cities around the world). I have verified the
         entries in the data section of the tests within +- 2 min. using
         the information from the Naval Observatory (aa.usno.navy.mil).

0.0401  Thurs Oct 8 2003
       - changed the _sunrise sub to include the timezone information
       - (I don't like the way I did this needs more research)

0.04   Fri May 30 7:40 2003
       - Fixed a bug in _sunrise
         The problem was I was setting the hours/min for the rise/set times
         I needed to create a duration and add the duration to the DateTime
         Object.

0.03   Tues April 29 14:00 2003
       - initial release to CPAN

0.02   Mon April 21 13:20:00 2003
        - doc tweaks. Flavio S. Glock
        - renamed sun_rise_set to sunrise_sunset
        - gets methods from DateTime::Set

0.01   Mon Mar 21 10:01:50 2003
        - original version; created by Ron Hill



Makefile.PL
#
#     Configuration script for DateTime::Event::Sunrise
#     Copyright © 2003, 2004, 2013, 2015, 2020 Ron Hill and Jean Forget
#
#     This program is distributed under the same terms as Perl 5.16.3:
#     GNU Public License version 1 or later and Perl Artistic License
#
#     You can find the text of the licenses in the F<LICENSE> file or at
#     L<https://dev.perl.org/licenses/artistic.html>
#     and L<https://www.gnu.org/licenses/gpl-1.0.html>.
#
#     Here is the summary of GPL:
#
#     This program is free software; you can redistribute it and/or modify
#     it under the terms of the GNU General Public License as published by
#     the Free Software Foundation; either version 1, or (at your option)
#     any later version.
#
#     This program is distributed in the hope that it will be useful,
#     but WITHOUT ANY WARRANTY; without even the implied warranty of
#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#     GNU General Public License for more details.
#
#     You should have received a copy of the GNU General Public License
#     along with this program; if not, write to the Free Software Foundation,
#     Inc., <https://www.fsf.org/>.
#
use ExtUtils::MakeMaker;
# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.
WriteMakefile(
    'NAME'              => 'DateTime::Event::Sunrise',
    'VERSION_FROM'      => 'lib/DateTime/Event/Sunrise.pm', # finds $VERSION
    'ABSTRACT'          => 'Perl DateTime extension for computing the sunrise/sunset on a given day',
    'PREREQ_PM'         => { 'DateTime::Set'    => '0.16',
                             'DateTime'         => '0.21',
                             'POSIX'            => '0',     # how can I find the minimum version without trying all of them? At least mention the dependency.
                             'Math::Trig'       => '0',
                             'Carp'             => '0',
                             'Params::Validate' => '0',
                             'Set::Infinite'    => '0',
                           },
    'AUTHOR'            => [ 'Ron Hill <rkhill@firstlight.net>', 'Jean Forget <JFORGET@cpan.org>' ],
    'LICENSE'           => 'perl',
    # The oldest Perl to check DT::E::Sunrise 0.0504 is 5.8.5. Therefore, I guess DT::E::Sunrise 0.0505 and next will work in 5.8.5 too.
    MIN_PERL_VERSION => '5.8.5',
    'BUILD_REQUIRES'    => { 'Test::More'         => '0',   # should be 'TEST_REQUIRES', except that ExtUtils::MakeMaker 6.5705 does not accept it
                             'DateTime::Duration' => '0',
                             'DateTime::Span'     => '0',
                             'DateTime::SpanSet'  => '0',
                             'Test::Exception'    => '0',
                             'ExtUtils::MakeMaker' => '6.57_02', # the first version to accept several authors in an arrayref
                           },
    META_MERGE       => {
       dynamic_config => 0,
       prereqs => {
         runtime => {
           requires => {
               'DateTime'           => '0.16',
               'DateTime::Duration' => '0',
               'DateTime::Span'     => '0',
               'DateTime::SpanSet'  => '0',
               'perl'               => '5.8.5',
           },
         },
         build => {
           requires => {
             'Test::More'          => '0',
             'ExtUtils::MakeMaker' => '6.57_02', # the first version to accept several authors in an arrayref
           },
         },
       },
       provides       => {
           'DateTime::Event::Sunrise' => {
                file => 'lib/DateTime/Event/Sunrise.pm',
                version => '0.0507',
           },
       },
       'meta-spec' => { version => 2 },
       resources => {
           repository => {
               type => 'git',
               url  => 'https://github.com/jforget/DateTime-Event-Sunrise.git',
               web  => 'https://github.com/jforget/DateTime-Event-Sunrise',
           },
       },
    },
);
